File: services/gameService/GameObject.java
package services.gameService;

import java.awt.Color;
import javax.swing.JPanel;

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
public abstract class GameObject extends JPanel{
    
    private GameSprite sprite;
    
    public GameObject(int x, int y, int width, int height, GameSprite sprite){
        this.sprite = sprite;
        if(sprite != null)
            add(this.sprite);
        
        setLocation(x, y);
        setSize(width, height);
        setLayout(null);
        setBackground(new Color(0, 0, 0, 1));
    }
    
    public void nextIndex(){
        sprite.nextIndex();
    }
    
    public void run(){
        nextIndex();
    }
    
}

File: services/gameService/GameObject.kt
package services.gameService

import java.awt.Color
import javax.swing.JPanel

/**
 *
 * @author Jean Carlos Santoya Cabrera jeancarlosodi@gmail.com
 */
abstract class GameObject(x: Int, y: Int, width: Int, height: Int, sprite: services.gameService.GameSprite?) : JPanel() {
    private val sprite: services.gameService.GameSprite?
    fun nextIndex() {
        if (sprite != null) {
            sprite.nextIndex()
        }
    }

    fun run() {
        nextIndex()
    }

    init {
        this.sprite = sprite
        if (sprite != null) add(this.sprite)
        setLocation(x, y)
        setSize(width, height)
        layout = null
        background = Color(0, 0, 0, 1)
    }
}